development/generate-rc-commits.js

```diff
@@ -157,17 +93,27 @@ async function filterCommitsByTeam(branchA, branchB) {
     const log = await git.log(logOptions);
     const seenMessages = new Set();
     const commitsByTeam = {};
+    let processedCommits = 0;
 
     const MAX_COMMITS = 500; // Limit the number of commits to process
+
     console.log('Generation of the CSV file "commits.csv" is in progress...');
 
     for (const commit of log.all) {
-      const { author, message, hash } = commit;
-      if (commitsByTeam.length >= MAX_COMMITS) {
+      if (processedCommits >= MAX_COMMITS) {
```

Comment:
Tested and found the current `commitsByTeam.length >= MAX_COMMITS`does not work because commitsByTeam is a object, not a Map, so it doesn't have a size property. We should track the number of processed commits by processedCommits.